Mathlib/Combinatorics/SimpleGraph/Metric.lean

```diff
@@ -128,6 +128,13 @@ theorem edist_eq_one_iff_adj : G.edist u v = 1 ↔ G.Adj u v := by
     exact w.adj_of_length_eq_one <| Nat.cast_eq_one.mp <| h ▸ hw
   · exact le_antisymm (edist_le h.toWalk) (ENat.one_le_iff_pos.mpr <| edist_pos_of_ne h.ne)
 
+lemma edist_bot [DecidableEq V] : (⊥ : SimpleGraph V).edist u v = (if u = v then 0 else ⊤) := by
```

Comment:
I think the "special" case of `edist u v = ⊤` in the case that u and v are distinct should be a lemma, in the same way that eg `Function.update_noteq` is. (The case that they're equal is fine since it's in `edist_self` already, but probably that should be a simp-lemma?)

Reply:
Oh sorry, I didn't mean to say you should get rid of `edist_bot`, instead that we should have both! `edist_bot_of_ne` is a fine name